packages/history/tests/createHashHistory.test.ts

```diff
@@ -0,0 +1,24 @@
+import { describe, expect, test } from 'vitest'
+
+import { createHashHistory } from '../src'
+
+describe('createHashHistory', () => {
+  test('parseLocation', () => {
```

Comment:
Please break this one large test function into individual ones.

```ts
describe('createHashHistory', () => {
	describe('parseLocation',  () => {
		test('...', () => {})

		test('...', () => {})

		// ...
	})
})
```

Maybe add some into a matrix where it checks both "on load's `parseLocation` call" and "after pushState's `parseLocation`" values.

Reply:
Sure. I'll break one test into smaller ones. Can you elaborate more on the other suggestion "add some into a matrix ...". I don't think I understand that.

Reply:
Maybe the word matrix wasn't the right word here. Essentially, what I'm talking about is checking the behaviours in both "on load" and "on navigate".

Something like this for example.

```ts
describe.each([
	['/', { pathname: '', search: '' }],
	// ...
])('check for %i', ([path, exp]) => {
	// check if parseLocation works in initial load
	test(`onLoad with ${path}`, () => {
		window.location.href=path
		const history = createHashHistory()
		expect(...).toBe(exp['...'])
	})

	// check if parseLocation works after navigate
	test(`onNavigate with ${path}`, () => {
		const history = createHashHistory()
		window.history.pushState({}, null, path)
		expect(...).toBe(exp['...'])
	})
})
```

Reply:
Oh okay, thanks for the explanation. Sure will work on that.